### PR TITLE
Virtual Router Service Offering

### PR DESCRIPTION
**Pull request by ustcweizhou**
content to be added

link to https://github.com/apache/cloudstack/pull/3946

### CODE REVIEW

**Comment by rhtyd** [comment]
ping @ustcweizhou are we expecting the content?

**Comment by rhtyd** [comment]
ping @ustcweizhou

**Comment by sureshanaparti** [comment]
Hi @ustcweizhou Please fix the conflicts. Are you adding the content, and will this be ready for 4.16.1?

File: source/adminguide/service_offerings.rst [hunk 1]
```diff
@@ -697,3 +697,8 @@ default system offering used for System VMs.
 #. Destroy the existing CPVM or SSVM offerings and wait for them to be
    recreated. The new CPVM or SSVM are configured with the new offering.
 
+
+Changing the Default System Offering for Virtual Routers
+---------------------------------------------------
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:702]: Add content please @ustcweizhou

**Comment by weizhouapache** [comment]
@blueorangutan docbuild

**Comment by weizhouapache** [comment]
@blueorangutan docbuild

**Comment by weizhouapache** [comment]
@blueorangutan docbuild

**Comment by weizhouapache** [comment]
@sureshanaparti this is ready for review

http://qa.cloudstack.cloud/docs/WIP-PROOFING/pr/157/adminguide/service_offerings.html#changing-the-default-system-offering-for-virtual-routers

File: source/adminguide/service_offerings.rst [hunk 2]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
 
 #. Destroy the existing CPVM or SSVM offerings and wait for them to be
    recreated. The new CPVM or SSVM are configured with the new offering.
+
+
+Changing the Default System Offering for Virtual Routers
+---------------------------------------------------
+
+Additionally, as a CloudStack administrator, you can change the
+default system offering used for Virtual Routers.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:730]:
```suggestion
As a CloudStack administrator, you can change the default system
offering used for Virtual Routers.
```

File: source/adminguide/service_offerings.rst [hunk 3]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+#. Create a new system Offering
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:732]:
```suggestion
#. Create a new system offering
```

File: source/adminguide/service_offerings.rst [hunk 4]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+   For more information, see `“Creating a New System Offering”
+   <service_offerings.html#creating-a-new-system-service-offering>`_.
+
+#. (Optional) Create a new network offering with System Offering
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:737]:
```suggestion
#. (Optional) Create a new network offering with SystemOffering
```

File: source/adminguide/service_offerings.rst [hunk 5]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+   For more information, see `“Creating a New Network Offering”
+   <networking.html#creating-a-new-network-offering>`_.
+
+#. (Optional) Change account setting
+
+   You can change the default system offering for Virtual Routers of a particular
+   account by changing account settings "router.service.offering" to uuid of system offering.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:745]:
```suggestion
   You can change the default system offering for Virtual Routers of a particular
   account by changing the account's setting "router.service.offering" to the uuid
   of the system offering.
```

File: source/adminguide/service_offerings.rst [hunk 6]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+   For more information, see `“Setting Local Configuration Parameters”
+   <../installguide/configuration.html#setting-local-configuration-parameters>`_.
+
+#. (Optional) Change global configuration
+
+   You can change the default system offering for Virtual Routers of all accounts
+   by changing global configuration "router.service.offering" to uuid of the system offering.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:753]:
```suggestion
   by changing the global configuration "router.service.offering" to the uuid of the system offering.
```

File: source/adminguide/service_offerings.rst [hunk 7]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+   For more information, see `“Setting Global Configuration Parameters”
+   <../installguide/configuration.html#setting-global-configuration-parameters>`_.
+
+When you create a network, the virtual routers will use the system offering in Network Offering.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:758]:
```suggestion
When you create a network, the virtual routers will use the system offering in their NetworkOffering.
```

File: source/adminguide/service_offerings.rst [hunk 8]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+If it is not set, the virtual routers will use system offering in account setting.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:759]:
```suggestion
If it is not set, the virtual routers will use the system offering in the account setting.
```

File: source/adminguide/service_offerings.rst [hunk 9]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+If account setting is not set, the virtual routers will use system offering in global configuration.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:760]:
```suggestion
If the account setting is not set, the virtual routers will use the system offering set in the global configuration.
```

File: source/adminguide/service_offerings.rst [hunk 10]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+If global configuration is not set, the virtual routers will use the default system offering for virtual 
+routers ("System Offering For Software Router" or "System Offering For Software Router - Local Storage").
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:762]:
```suggestion
If the global configuration is not set, the virtual routers will use the default system offering for virtual 
routers ("System Offering For Software Router" or "System Offering For Software Router - Local Storage").
```

File: source/adminguide/service_offerings.rst [hunk 11]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+
+You can update an existing network to a new network offering. The new virtual routers will use the 
+new system offering set in Network Offering, account setting or global configuration.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:765]:
```suggestion
You can update an existing network to a new network offering. The new virtual routers will use the 
new system offering set in the NetworkOffering, account setting or global configuration.
```

File: source/adminguide/service_offerings.rst [hunk 12]
```diff
@@ -721,3 +721,52 @@ default system offering used for System VMs.
...
+For more information, see `“Changing the Network Offering on a Guest Network”
+<networking_and_traffic.html#changing-the-network-offering-on-a-guest-network>`_.
+
+You can restart the network with cleanup. The new virtual routers will use the new system offering
+set in Network Offering, account setting or global configuration.
```
Comment (DaanHoogland) [inline source/adminguide/service_offerings.rst:770]:
```suggestion
You can restart the network with cleanup. The new virtual routers, created after the restart, will use
the new system offering, set in the NetworkOffering, account setting or global configuration.
```

File: source/adminguide/networking/manage_guest_networks.rst [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+
+Editing, Restarting, and Removing a Guest Network
+--------------------------------------------------
+
+.. note:: Ensure that all the vms are removed before you remove a guest network.
```
Comment (DaanHoogland) [inline source/adminguide/networking/manage_guest_networks.rst:20]:
```suggestion
.. note:: Ensure that all the VMs are removed before you remove a guest network.
```

File: source/adminguide/networking/manage_guest_networks.rst [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+#. Log in to the CloudStack UI as an administrator or end user.
+
+#. In the left navigation, choose Network.
+
+#. In the Select view, select Guest networks.
+
+   All the guest networks that you have created for the account is listed in the
+   page.
```
Comment (DaanHoogland) [inline source/adminguide/networking/manage_guest_networks.rst:29]:
```suggestion
   All the guest networks that you have created for the account are listed in the
   page.
```

File: source/adminguide/networking/manage_guest_networks.rst [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+#. Select the guest network you want to work with.
+
+#. In the Details tab, click the "Delete Network" button
+
+   You can remove the guest network by also using the remove button in the Quick
+   View.
```
Comment (DaanHoogland) [inline source/adminguide/networking/manage_guest_networks.rst:36]:
```suggestion
   You can also remove the guest network by using the remove button in the Quick
   View.
```

**Review by DaanHoogland — APPROVED** [review]
nitpicking about "lidwoorden" @weizhouapache , but looks good

**Comment by weizhouapache** [comment]
thanks @DaanHoogland for view !
pr is updated

**Comment by weizhouapache** [comment]
@blueorangutan docbuild

File: source/adminguide/networking/manage_guest_networks.rst [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+#. In the Details tab, click the "Delete Network" button
+
+   You can also remove the guest network by using the remove button in the Quick
+   View.
+
+   You can edit the name, description, network offering, CIDR, network domain of a 
+   guest network. To do that, click the "Edit" button.
+
+   To restart a guest network, click the "Restart network" button. Please note 
+   all services provided by this network will be interrupted. When you check "Clean up",
```
Comment (sureshanaparti) [inline source/adminguide/networking/manage_guest_networks.rst:42]:
```suggestion
   all services provided by this network will be interrupted. When you enable "Clean up",
```

File: source/adminguide/service_offerings.rst [hunk 17]
```diff
@@ -721,3 +721,53 @@ default system offering used for System VMs.
...
    recreated. The new CPVM or SSVM are configured with the new offering.
+
+
+Changing the Default System Offering for Virtual Routers
+---------------------------------------------------
+
+As a CloudStack administrator, you can change the default system
+offering used for Virtual Routers.
+
+#. Create a new system offering
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:732]:
```suggestion
#. Create a new system service offering
```

File: source/adminguide/service_offerings.rst [hunk 18]
```diff
@@ -721,3 +721,53 @@ default system offering used for System VMs.
...
+
+#. (Optional) Change global configuration
+
+   You can change the default system offering for Virtual Routers of all accounts
+   by changing the global configuration "router.service.offering" to the uuid of the system offering.
+
+   For more information, see `“Setting Global Configuration Parameters”
+   <../installguide/configuration.html#setting-global-configuration-parameters>`_.
+
+When you create a network, the virtual routers will use the system offering in their NetworkOffering.
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:759]:
```suggestion
When you create a network, the virtual routers will use the system service offering in their Network Offering.
```

File: source/adminguide/service_offerings.rst [hunk 19]
```diff
@@ -721,3 +721,53 @@ default system offering used for System VMs.
...
+If the global configuration is not set, the virtual routers will use the default system offering for virtual 
+routers ("System Offering For Software Router" or "System Offering For Software Router - Local Storage").
+
+You can update an existing network to a new network offering. The new virtual routers will use the 
+new system offering set in the NetworkOffering, account setting or global configuration.
+For more information, see `“Changing the Network Offering on a Guest Network”
+<networking_and_traffic.html#changing-the-network-offering-on-a-guest-network>`_.
+
+You can restart the network with cleanup. The new virtual routers, created after the restart, will use
+the new system offering, set in the NetworkOffering, account setting or global configuration.
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:771]:
```suggestion
the new system offering, set in the Network Offering, account setting or global configuration.
```

File: source/adminguide/service_offerings.rst [hunk 20]
```diff
@@ -721,3 +721,53 @@ default system offering used for System VMs.
...
+   <../installguide/configuration.html#setting-global-configuration-parameters>`_.
+
+When you create a network, the virtual routers will use the system offering in their NetworkOffering.
+If it is not set, the virtual routers will use the system offering in the account setting.
+If the account setting is not set, the virtual routers will use the system offering set in the global configuration.
+If the global configuration is not set, the virtual routers will use the default system offering for virtual 
+routers ("System Offering For Software Router" or "System Offering For Software Router - Local Storage").
+
+You can update an existing network to a new network offering. The new virtual routers will use the 
+new system offering set in the NetworkOffering, account setting or global configuration.
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:766]:
```suggestion
new system offering set in the Network Offering, account setting or global configuration.
```

**Comment by weizhouapache** [comment]
@blueorangutan docbuild

**Review by sureshanaparti — APPROVED** [review]

File: source/adminguide/service_offerings.rst (hunk 18 above)
Comment (DaanHoogland) [reply source/adminguide/service_offerings.rst]: I don't agree with this, either we should use camel case or we should use english concepts, not a mix of both.

**Pull request merged**